patterns/state-scoped-entities/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+use bevy::prelude::*;
+
+pub struct StateScopedEntitiesPlugin;
+
+impl Plugin for StateScopedEntitiesPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_state::<MyState>();
+        app.add_systems(Update, despawn_scoped::<MyState>);
```

Comment:
Maybe `PreUpdate`? That way Commands get applied before most user systems run, but still after state transitions

Reply:
Good idea!